csi.proto

```diff
@@ -1257,6 +1290,35 @@ message VolumeUsage {
   // Units by which values are measured. This field is REQUIRED.
   Unit unit = 4;
 }
+
+message VolumeHealth {
+  enum Severity {
+    // Volume health condition is unknown; treat it as healthy.
+    UNKNOWN = 0;
+    // Volume is healthy.
+    HEALTHY = 1;
+    // Volume is temporarily unhealthy, but will likely become healthy.
+    TEMPORARY_UNHEALTHY = 2;
+    // Volume will fail soon. Move your data off and replace volume or
+    // you may lose your data.
+    FATAL_EMINENT = 3;
+    // Volume is unusable. Data loss is likely.
+    FATAL = 4;
+  }
+
+  // The severity level of the health condition of the volume.
+  // This field is REQUIRED.
+  Severity severity = 1;
+
+  // The error code describing the health condition of the volume.
+  // This is an opaque field to CO. Only SP knows how to interprete it.
```

Comment:
I'd suggest removing the sentence "Only SP knows how to interprete it."  Given that the CO will not pass this information back to the SP anywhere in the API, I imagine that it will get reported to the end user, who may also know how to interpret the information.  Also, if you feel that this sentence is important and must be included, note that the spelling of "interpret" is incorrect.

Reply:
Sure.  I removed it.